UITestHelpers/XCUIElement+Extensions.swift

```diff
@@ -0,0 +1,32 @@
+import XCTest
+
+extension XCUIElement {
+
+    public func paste(text: String) -> Void {
+        let previousPasteboardContents = UIPasteboard.general.string
+        UIPasteboard.general.string = text
+
+        self.press(forDuration: 1.2)
+        XCUIApplication().menuItems.firstMatch.tap()
+
+        if let string = previousPasteboardContents {
+            UIPasteboard.general.string = string
+        }
+    }
+
+    public func clearAndEnterText(text: String) {
+        self.clearText()
+        self.typeText(text)
+    }
+
+    public func clearText() {
+        guard let stringValue = self.value as? String else {
+            XCTFail("Tried to clear and enter text into a non string value")
+            return
+        }
```

Comment:
I like this pre-check! I think we should use it wherever possible.

I'm not sure why Apple decided to put everything behind the `XCUIElement` proxy instead of having a more refined type structure 🤔 ...

I'm also not sure failing the test from here is the best approach in the context of a library. That's a discussion that's out of scope for this PR, though. I opened https://github.com/Automattic/simplenote-ios/issues/1244 to talk about it.

Reply:
I fully support not having assertions anywhere besides asserting methods 👍 Thank you for the issue

Reply:
> I like this pre-check! I think we should use it wherever possible.

The whole initial method is unfortunately not mine... I just added the credits. I added it during my trial, when I was super time-limited, and then forgot to do this after

Reply:
No worries. I'm a fan of getting things in fast, then refining them. As Kent Beck says, _Make it work, make it right, make it fast_.